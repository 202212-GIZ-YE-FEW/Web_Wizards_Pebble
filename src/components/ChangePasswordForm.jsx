import EditprofileButton from "./EditProfileButton";
import InputForm from "./InputForm";
import SaveButton from "./SaveButton";
import SecondaryTitle from "./SecondaryTitle";
const ChangePasswordForm = ({ t }) => {
    return (
        <div className='w-11/12 mx-auto rounded-3xl flex flex-col justify-center bg-[#d4f3fa] md:px-10 px-2 py-10'>
            <SecondaryTitle text={t("changePassword")} />
            <div className='flex flex-row py-6'>
                <InputForm
                    placeholder={t("newPassword")}
                    label={t("newPassword")}
                    name='New Password'
                    type='Password'
                    hasFloatingLabel
                    togglePasswordLabel='Show/Hide password'
                    togglePassword='true'
                />
                <InputForm
                    placeholder={t("confirmPassword")}
                    label={t("confirmPassword")}
                    name='Confirm Password'
                    type='Password'
                    hasFloatingLabel
                    togglePasswordLabel='Show/Hide password'
                    togglePassword='true'
                />
            </div>

            <div className='flex flex-row '>
                <EditprofileButton text={t("cancel")} />
                <SaveButton text={t("submit")} />
            </div>
        </div>
    );
};

export default ChangePasswordForm;
