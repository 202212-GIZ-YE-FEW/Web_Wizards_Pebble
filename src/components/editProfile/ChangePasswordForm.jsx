import { useState } from "react";

import EditprofileButton from "@/components/editProfile/EditProfileButton";
import SaveButton from "@/components/editProfile/SaveButton";
import SecondaryTitle from "@/components/editProfile/SecondaryTitle";
import InputForm from "@/components/shared/InputForm";

import { changePassword } from "@/firebase/auth";

async function handlePasswordChange(e) {
    const form = e.target;
    const formData = new FormData(form);
    const password = formData.get("password");
    const confirmPassword = formData.get("passwordConfirm");
    if (password != confirmPassword) return;
    await changePassword(password);
}

const ChangePasswordForm = ({ t }) => {
    const [passwordLoading, setPasswordLoading] = useState(false);
    return (
        <div className='w-11/12 mx-auto rounded-3xl flex flex-col justify-center bg-[#d4f3fa] md:px-10 px-2 py-10'>
            <SecondaryTitle text={t("changePassword")} />
            <form
                onSubmit={async (e) => {
                    e.preventDefault();
                    setPasswordLoading(true);
                    await handlePasswordChange(e);
                    setTimeout(() => {
                        setPasswordLoading(false);
                    }, 1000);
                    e.target.reset();
                }}
            >
                <div className='flex flex-row py-6'>
                    <InputForm
                        placeholder={t("newPassword")}
                        label={t("newPassword")}
                        name='password'
                        type='password'
                        hasFloatingLabel
                        togglePasswordLabel='Show/Hide password'
                        togglePassword='true'
                    />
                    <InputForm
                        placeholder={t("confirmPassword")}
                        label={t("confirmPassword")}
                        name='confirmPassword'
                        type='password'
                        hasFloatingLabel
                        togglePasswordLabel='Show/Hide password'
                        togglePassword='true'
                    />
                </div>
                <div className='flex flex-row justify-end'>
                    <input type='submit' id='changePasswordSubmit' hidden />
                    <SaveButton
                        text={t("submit")}
                        htmlFor='changePasswordSubmit'
                        loading={passwordLoading}
                    />
                    <EditprofileButton
                        text={t("cancel")}
                        className='!w-36 !m-0 !ml-2'
                    />
                </div>
            </form>
        </div>
    );
};

export default ChangePasswordForm;
