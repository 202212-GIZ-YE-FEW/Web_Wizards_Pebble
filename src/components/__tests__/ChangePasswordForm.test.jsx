import { I18nContext } from "next-i18next";
import renderer from "react-test-renderer";

import ChangePasswordForm from "../ChangePasswordForm";

it("renders correctly", () => {
    const i18n = {
        language: "en",
        t: (str) => str,
    };
    const tree = renderer
        .create(
            <I18nContext.Provider value={i18n}>
                <ChangePasswordForm t={i18n.t} />
            </I18nContext.Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
