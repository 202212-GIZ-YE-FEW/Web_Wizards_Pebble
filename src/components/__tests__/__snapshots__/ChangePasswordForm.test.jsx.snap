// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-11/12 mx-auto rounded-3xl flex flex-col justify-center bg-[#d4f3fa] md:px-10 px-2 py-10"
>
  <div>
    <label
      className="font-extrabold text-2xl py-2 ml-3 text-gray-800"
    >
      changePassword
    </label>
  </div>
  <div
    className="flex flex-row py-6"
  >
    <div
      className="px-3 mb-3 w-full "
    >
      <div
        className="input__InputWrapper-sc-1hj52gj-0 hyTiyQ input-wrapper floating-label toggle-password"
      >
        <input
          className="text-input default !bg-white !border-2 !border-[black] !rounded-lg !shadow-none focus:[&:not(#password)]:!border-[3px] focus:[&:not(#password)]:!border-[black] focus:ring-0"
          id="f8445d83-d755-5b"
          name="New Password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          pattern="^(?=.*[A-Z])(?=.*\\\\d)[A-Za-z\\\\d]{8,32}$"
          placeholder="newPassword"
          required={false}
          type="password"
          value=""
        />
        <label
          className="form-label"
          htmlFor="f8445d83-d755-5b"
        >
          newPassword
        </label>
        <button
          aria-label="Show/Hide password"
          className="show-password-button"
          onClick={[Function]}
        />
      </div>
      <p
        className="italic"
      />
    </div>
    <div
      className="px-3 mb-3 w-full "
    >
      <div
        className="input__InputWrapper-sc-1hj52gj-0 hyTiyQ input-wrapper floating-label toggle-password"
      >
        <input
          className="text-input default !bg-white !border-2 !border-[black] !rounded-lg !shadow-none focus:[&:not(#password)]:!border-[3px] focus:[&:not(#password)]:!border-[black] focus:ring-0"
          id="9b21eda7-1b1c-54"
          name="Confirm Password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          pattern="^(?=.*[A-Z])(?=.*\\\\d)[A-Za-z\\\\d]{8,32}$"
          placeholder="confirmPassword"
          required={false}
          type="password"
          value=""
        />
        <label
          className="form-label"
          htmlFor="9b21eda7-1b1c-54"
        >
          confirmPassword
        </label>
        <button
          aria-label="Show/Hide password"
          className="show-password-button"
          onClick={[Function]}
        />
      </div>
      <p
        className="italic"
      />
    </div>
  </div>
  <div
    className="flex flex-row "
  >
    <div
      className=" w-1/3 flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 sm:py-1 sm:px-2  "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button w-full h-16 px-1 transition "
        disabled={false}
        style={
          Object {
            "backgroundColor": "white",
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center space-x-4 justify-center"
        >
          <span
            className="block w-max font-semibold tracking-wide text-gray-900 sm:text-base text-sm md:text-lg lg:text-xl "
          >
            cancel
          </span>
        </div>
      </button>
    </div>
    <div
      className="px-3 py-2 lg:py-4 "
    >
      <button
        className="w-36 h-16 lg:w-56 bg-primary-200  text-white font-semibold rounded-lg text-sm lg:text-md px-3 sm:px-5 py-3  undefined"
      >
        submit
      </button>
    </div>
  </div>
</div>
`;
